source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,190 @@
+.. _fundamentals-kotlin-serialization:
+
+====================
+Kotlin Serialization
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+The Kotlin driver supports the ``kotlinx.serialization`` library for serializing 
+and deserializing Kotlin objects. 
+
+The driver provides an efficient ``Bson`` serializer that can can be used with 
+``@Serializable`` classes to handle the serialization of Kotlin objects to BSON data. 
+The ``bson-kotlinx`` library also supports :ref:`custom codecs <kotlin-custom-codec>`
+with configurations such as encoding defaults, encoding nulls, and defining class 
+discriminators.
+
+.. note::
+
+    To use the ``Codec`` interface to specify custom encoding and decoding of 
+    Kotlin objects to BSON data, see the :ref:`Codecs <fundamentals-codecs>` 
+    documentation.
+
+Although you can use the Kotlin driver with the Kotlin Serialization's ``Json`` 
+library, the ``Json`` serializer does *not* directly support BSON value types such
+as ``ObjectId``. You must provide a custom serializer that can handle the 
+conversion between BSON and JSON. 
+
+Supported Types
+~~~~~~~~~~~~~~~
+
+The Kotlin driver supports all Kotlin types that are supported by the Kotlin 
+Serialization library and the following BSON types:
+
+- ObjectId
+- BsonNull
+- BsonArray
+- BsonBinary
+- BsonBoolean
```

Comment:
Q: Is there a list you can link to instead of having all of them on this page?

Reply:
updated to link out to https://www.mongodb.com/docs/manual/reference/bson-types/